hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/command/CreateHoodieTableLikeCommand.scala

```diff
@@ -0,0 +1,112 @@
+import org.apache.hudi.common.util.ConfigUtils
+import org.apache.spark.sql.{AnalysisException, Row, SparkSession}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.catalyst.catalog.{CatalogStorageFormat, CatalogTable, CatalogTableType, HoodieCatalogTable}
+import org.apache.spark.sql.catalyst.util.CharVarcharUtils
+import org.apache.spark.sql.hudi.HoodieOptionConfig
+
+import scala.util.control.NonFatal
+
+case class CreateHoodieTableLikeCommand(targetTable: TableIdentifier,
+                                        sourceTable: TableIdentifier,
+                                        fileFormat: CatalogStorageFormat,
+                                        properties: Map[String, String] = Map.empty,
+                                        ignoreIfExists: Boolean)
+  extends HoodieLeafRunnableCommand with SparkAdapterSupport {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+
+    val tableIsExists = catalog.tableExists(targetTable)
+    if (tableIsExists) {
+      if (ignoreIfExists) {
+        // scalastyle:off
+        return Seq.empty[Row]
+        // scalastyle:on
+      } else {
+        throw new IllegalArgumentException(s"Table $targetTable already exists.")
+      }
+    }
+
+    val sourceTableDesc = catalog.getTempViewOrPermanentTableMetadata(sourceTable)
+
+    val newStorage = if (fileFormat.inputFormat.isDefined) {
+      fileFormat
+    } else {
+      sourceTableDesc.storage.copy(locationUri = fileFormat.locationUri)
+    }
+
+    // If the location is specified, we create an external table internally.
+    // Otherwise create a managed table.
+    val tblType = if (newStorage.locationUri.isEmpty) {
+      CatalogTableType.MANAGED
+    } else {
+      CatalogTableType.EXTERNAL
+    }
+
+    val targetTableProperties = if (sparkAdapter.isHoodieTable(sourceTableDesc)) {
+      HoodieOptionConfig.extractHoodieOptions(sourceTableDesc.properties) ++ properties
+    } else {
+      properties
+    }
+
+    val newTableSchema = CharVarcharUtils.getRawSchema(sourceTableDesc.schema)
+    val newTableDesc = CatalogTable(
+      identifier = targetTable,
+      tableType = tblType,
+      storage = newStorage,
+      schema = newTableSchema,
+      provider = Some("hudi"),
+      partitionColumnNames = sourceTableDesc.partitionColumnNames,
+      bucketSpec = sourceTableDesc.bucketSpec,
+      properties = targetTableProperties,
+      tracksPartitionsInCatalog = sourceTableDesc.tracksPartitionsInCatalog)
+
+    val hoodieCatalogTable = HoodieCatalogTable(sparkSession, newTableDesc)
+    // check if there are conflict between table configs defined in hoodie table and properties defined in catalog.
+    CreateHoodieTableCommand.validateTblProperties(hoodieCatalogTable)
+
+    val queryAsProp = hoodieCatalogTable.catalogProperties.get(ConfigUtils.IS_QUERY_AS_RO_TABLE)
+    if (queryAsProp.isEmpty) {
```

Comment:
Does use specify this option through sql options?

Reply:
From my internal usage, we don't see users manually specify it, here I just want to align with `CreateHoodieTableCommand`, which checks this property.